qualtran/bloqs/multi_control_multi_target_pauli.py

```diff
@@ -137,6 +140,160 @@ def _has_unitary_(self) -> bool:
         return True
 
 
+@frozen
+class MultiControlX(Bloq):
+    r"""Implements multi-control, single-target X gate as a bloq using $n-1$ clean ancillas.
+
+    References:
+        [Constructing Large Controlled Nots]
+        (https://algassert.com/circuits/2015/06/05/Constructing-Large-Controlled-Nots.html)
+    """
+
+    cvs: Tuple[int, ...] = field(converter=lambda v: (v,) if isinstance(v, int) else tuple(v))
+
+    @cached_property
+    def signature(self) -> 'Signature':
+        if len(self.cvs) > 0:
+            return Signature([Register('ctrl', bitsize=len(self.cvs)), Register('x', bitsize=1)])
+        else:
+            return Signature([Register('x', bitsize=1)])
+
+    def on_classical_vals(
+        self, x: 'ClassicalValT', **vars: 'ClassicalValT'
+    ) -> Dict[str, 'ClassicalValT']:
+        if len(self.cvs) > 0:
+            ctrl = vars['ctrl']
+        else:
+            return {'x': (x + 1) % 2}
+
+        ctrls = bits_to_ints(self.cvs)
+        if ctrls == ctrl:
```

Comment:
here's another philosophical thing: we support both

```python
Register('ctrl', bitsize=5)
Register('ctrl', bitsize=1, shape=(5,)
```

the former is one 5-bit-wide wire whereas the latter is a 1d array of 1-bit wires. If you want `cvs` to be a sequence of bits and you want this to be called `MultiControlledX` then you should probably use the second construction (and remove the `bits_to_ints` call inside this method). If you want `cvs` to be one control integer, use the first one, call it `IntControlledX`, and remove the `bits_to_ints` call here.